pdns/pdnsutil.cc

```diff
@@ -1761,8 +1761,12 @@ static int addOrReplaceRecord(bool isAdd, const vector<string>& cmds) {
   DNSName name;
   if (cmds.at(2) == "@")
     name=zone.operator const DNSName&();
-  else
-    name = DNSName(cmds.at(2)) + zone.operator const DNSName&();
+  else {
+    name = DNSName(cmds.at(2));
+    if (!name.isPartOf(zone)) {
+      throw PDNSException("Name to add is not part of zone.");
```

Comment:
Adding the name of the record and zone here will improve the clarity of the error.

Reply:
Oh wow, it seems my C++ knowledge is good enough to guess how this can be done :-)